src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -1246,14 +1246,16 @@ FCIMPL1(void*, RuntimeFieldHandle::GetStaticFieldAddress, ReflectFieldObject *pF
     // IsFastPathSupported needs to checked before calling this method.
     _ASSERTE(IsFastPathSupportedHelper(pFieldDesc));
 
-    PTR_BYTE base = 0;
-    if (!pFieldDesc->IsRVA())
+    if (pFieldDesc->IsRVA())
     {
-        // For RVA the base is ignored and offset is used.
-        base = pFieldDesc->GetBase();
+        Module* pModule = pFieldDesc->GetModule();
+        return pModule->GetRvaField(pFieldDesc->GetOffset());
+    }
+    else
+    {
+        PTR_BYTE base = pFieldDesc->GetBase();
+        return PTR_VOID(base + pFieldDesc->GetOffset());
     }
-
-    return PTR_VOID(base + pFieldDesc->GetOffset());
 }
```

Comment:
This is actually a bugfix for the fast field access feature.
Currently in main the following code will cause access violation:
```csharp
            foreach (var type in typeof(Program).Assembly.GetTypes())
            {
                if (type.Name.Contains("PrivateImplementationDetails"))
                {
                    foreach (var field in type.GetFields(System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic))
                    {
                        _ = field.GetValue(null);
                        _ = field.GetValue(null);
                    }
                }
            }
```

Reply:
Well the `FieldAccessor` doesn't support RVA field at all. It gets correct address now but attempts to read it as a box.

Reply:
I'd like to reuse this FCall for its semantic. The test will remain broken for now, and I will submit a PR to fix it.

Reply:
Could you please include regression test for this and open an issue to track the bug?

Reply:
I'm already working on the fix with this. If the regression test is included in this PR, should it be disabled with `OpenIssue`?

Reply:
Yes, that would be good.

Reply:
Created #103207